### PR TITLE
810 searchbar component

### PR DESCRIPTION
**Pull request by Jacobjeevan**
#823 
 +

- added logic to rerender when user clicks on any of the filtering options (rather than having to click submit).
- only display agencies with wishcards in the agency sort dropdown (and ordered them alphabetically)
- Added no results message for now (maybe do additional styling?)

image

Note: Still has to use submit button for the text input ofc.

### CODE REVIEW

File: js/react/pages/wishCards.jsx [hunk 1]
```diff
@@ -7,57 +9,181 @@ import MantineProviderWrapper from '../utils/mantineProviderWrapper.jsx';
...
+				agencyFilter: searchQueryParams.agencyFilter
+					? agencies.find((agency) => agency.agencyName == searchQueryParams.agencyFilter)
+							._id
+					: null,
+			},
+			{
+				headers: {
+					'Content-Type': 'application/json',
+				},
+			},
```
Comment (Enubia) [inline js/react/pages/wishCards.jsx:84]: headers can be omitted if it's not a formdata object or you need to include some special headers for the request

File: js/react/pages/wishCards.jsx [hunk 2]
```diff
@@ -7,57 +9,181 @@ import MantineProviderWrapper from '../utils/mantineProviderWrapper.jsx';
...
 			<div id="wishcards" className="bg-light p-4">
-				<div className="container">
+				<div className="container d-flex flex-column">
+					<div className="d-flex justify-content-center">
+						<form className="p-4 rounded-2 bg-white" onSubmit={handleSearchSubmit}>
+							<div className="d-flex flex-wrap gap-3 mb-2">
+								<TextInput
+									placeholder="Search"
+									value={searchTextInput}
+									onChange={handleSearchInputChange}
```
Comment (Enubia) [inline js/react/pages/wishCards.jsx:99]: instead of an `onChange` event you could use a `ref` here so you don't fire an event everytime the input changes

**Review by Enubia — CHANGES_REQUESTED** [review]

File: js/react/pages/wishCards.jsx (hunk 2 above)
Comment (Jacobjeevan) [reply js/react/pages/wishCards.jsx]: Done

**Review by Enubia — APPROVED** [review]
LGTM!

**Pull request merged**